comm/zmq_sender.go

```diff
@@ -0,0 +1,38 @@
+package comm
+
+import (
+	zmq "github.com/pebbe/zmq4"
+)
+
+// ZMQSender - Data type that contains the zmq.Socket interface used for the connection
+type ZMQSender struct {
+	socket *zmq.Socket
+}
+
+// NewSocket connect to the specified socket for sending data
+func (z *ZMQSender) NewSocket(addr string) error {
+	zmq.SetRetryAfterEINTR(true)
+
+	socket, err := zmq.NewSocket(zmq.PAIR)
+	if err != nil {
+		return err
+	}
+
+	err = socket.Connect(addr)
+	if err != nil {
+		return err
+	}
+
+	z.socket = socket
+	return nil
+}
```

Comment:
Vou comentar em todos para tu poder te guiar por aqui. Aqui também é o mesmo esquema. Vale ressaltar que tudo isso que to indicando resulta em uma legibilidade praticamente perfeita do nosso código.